Tools/GraphGeneration/README.md

```diff
@@ -0,0 +1,67 @@
+# Graph Generation for NG911 Simulations
+
+## Overview
+This repository provides scripts for generating and visualizing synthetic star graphs representing Public Safety Answering Points (PSAPs), responders, and callers. These graphs are designed for simulations and analysis within emergency response systems. The generated graphs can be visualized using `matplotlib` and saved in GraphML format for further use.
+
+## Purpose
+The main objective is to create synthetic star graphs representing emergency response systems with:
+- PSAPs at the center.
+- Responders and callers distributed within specified regions.
+- Special handling for prime numbers of PSAPs and other specific cases.
+- The `visualize_generation.py` script specifically focuses on visualizing the generation of a synthetic network graph from parameterized GraphML data, highlighting different node types and their positions within defined regions.
+
+## graph_generation folder 
+
+### `graph_generation.py`
+Generates synthetic star graphs based on input parameters. The script handles various scenarios, including special cases such as prime numbers of PSAPs, to ensure appropriate distribution of graph elements. It includes logic for:
+ - Distributing PSAPs, responders, and callers within defined regions.
+ - Handling prime numbers of PSAPs by creating an extra region and combining it appropriately.
+ - Ensuring each region contains a star-shaped graph centered on a PSAP.
+
+### `visualize_generation.py`
+Visualizes the graphs generated by `graph_generation.py` from the GraphML file. It displays the graph with color-coded nodes for PSAPs, callers, and responders. It outlines the rectangular boundary, PSAP regions, and caller regions, allowing for easy analysis and verification of the graph structure.
+
+### `format_graphml.py`
+Converts the original synthetic GraphML file to a format that can be used within the cluster-point call generation and Graphitti simulations. This script ensures the GraphML file is compatible with the tools and configurations used in subsequent simulation steps.
+
+## call_generation folder 
+Contains scripts for generating call data from the GraphML files created in the `graphml_generation` folder. These scripts use the structured graphs to simulate call events, providing data for emergency response simulations.
```

Comment:
Why is there more than one Python file? They are simply numbered, which doesn't really explain their functions.

Reply:
Oh, and at second glance these aren't Python files. So, what are they? They aren't scripts.